Sofware Management Plan/README.md

```diff
@@ -0,0 +1,137 @@
+# Software Management Plan
+
+A sofware management plan should be provided for all projects which contribute to the ECMWF software stack.
+This is to ensure that contributions are appropriate, visible, follow guidelines and there is a clear roadmap
+for development and ownership.
+
+## 1. Project Overview
+
+Provide a concise overview to allow others to understand the project and its purpose.
+
+Include:
+
+- **Project name**
+- **Purpose and goals**
+- **Problem it solves / value it provides**
+- **Scope and boundaries** (what is included and excluded)
+- **Relationship to the ECMWF software stack**  
+  - New components?
+  - Extension?
+  - Replacement of legacy functionality?
+
+---
+
+## 2. Alignment With ECMWF Strategy and Architecture
+
+Explain how the software fits into ECMWF's broader technical and organisational goals.
+
+Address:
+
+- **Alignment with ECMWF technical strategy**
+- **Fit with ECMWF architecture, APIs, data formats, workflows, and coding standards**
+- **Dependencies** (internal and external)
+- **Avoiding duplication**
+- **Interoperability and portability**
+
+---
+
+## 3. Stakeholders and Ownership
+
+Identify roles and responsibilities.
+
+Include:
+
+- **Project owner / maintainer**
+- **Contributors / development team**
+- **Users and beneficiaries**
+- **Domain experts / reviewers**
+- **Points of contact**
+- **Long-term maintenance owner**
+
+---
+
+## 4. Development, Maintenance and Ownership Roadmap
```

Comment:
I suppose my question about this section is related to the fact that most of the other content would be fairly static, whereas this section could have some quite dynamic content, especially with agile development, and would therefore need regular updates - I'm thinking of planned features and timelines in particular. So I wonder if this section really belongs in this document, at least in its current form?

Reply:
I think this is one of the most important aspects of the SMP. Providing a framework for delivering software and clarifying who is the owner of that software when it is delivered is one of the big questions when working with these external contributions. There is not a central answer for this, and what I would like this process to make clear to all involved, e.g. Development section reps and the technical Technical Officers.

I agree that we can change from it's current form if needed, and maybe it's less relevant to internal projects and we could labels it as "only required for External Contributions"?